src/main/java/com/faforever/api/challonge/ChallongeController.java

```diff
@@ -75,12 +75,12 @@ public CompletableFuture<ResponseEntity<String>> write(@RequestBody(required = f
   }
 
   @RequiredArgsConstructor
-  private static class ChallongeUriTemplateHandler extends DefaultUriTemplateHandler {
+  private static class ChallongeUriTemplateHandler extends DefaultUriBuilderFactory {
     private final String apiKey;
 
     @Override
-    protected UriComponentsBuilder initUriComponentsBuilder(String uriTemplate) {
-      return super.initUriComponentsBuilder(uriTemplate).queryParam("api_key", apiKey);
+    public UriBuilder uriString(String uriTemplate) {
```

Comment:
not sure about this implementation. if anybody know how to test pls test or tell me :)

Reply:
looks like it is working